crypto/s390xcap.c

```diff
@@ -217,8 +217,17 @@ void OPENSSL_cpuid_setup(void)
         OPENSSL_s390xcex = -1;
     } else {
         OPENSSL_s390xcex = open("/dev/z90crypt", O_RDWR | O_CLOEXEC);
+        /*
+         * leaving atexit() here because library destructors are not
+	 * invoked when application is linked statically with libcrypto.
+	 * I'm not able to judge what happen if device is not closed
+	 * explicitly by application.
+	 *
+	 * On the other hand 21st century OS  closes all file descriptors
+	 * when application exits.
+         */
```

Comment:
This is a 21st century OS target - S390x Linux. Could you perhaps call this from OPENSSL_cleanup()?